admin/projects.go

```diff
@@ -434,3 +517,23 @@ func (s *Service) TriggerParserAndAwaitResource(ctx context.Context, depl *datab
 		}
 	}
 }
+
+func (s *Service) ResolveProdVariables(ctx context.Context, projectID string) (map[string]string, error) {
+	vars, err := s.DB.FindProjectVariablesByEnvironment(ctx, projectID, nil)
+	if err != nil {
+		return nil, err
+	}
+	res := make(map[string]string)
+	for _, v := range vars {
+		switch v.Environment {
+		case "":
+			// only add default variable if it doesn't already exist to not override production variables
+			if _, ok := res[v.Name]; !ok {
+				res[v.Name] = string(v.Value)
+			}
+		case "production":
+			res[v.Name] = string(v.Value)
+		}
```

Comment:
1. Should this do case insensitive comparisons? 
2. The environment is called `"prod"`, not `"production"`: https://github.com/rilldata/rill/blob/2396d5c36384a911502db944d3f7a4142738c5d1/admin/deployments.go#L173

Reply:
This switches on the environment set in the variable. I had thought it is `prod` and `dev`. I can change it.
Instead of doing case insensitive comparisons we can set environment in lower case only while inserting.

Reply:
Sorry I meant case insensitive comparisons for `v.Name` (since it uses `lower` in the database schema for uniqueness)